src/ex2/Uilogic.js

```diff
@@ -0,0 +1,444 @@
+const TYPE_DEADLINE = 1
+const TYPE_NOTCOMPLETE = 2
+const TYPE_FINISHED = 3
+const TYPE_DEADLINE_FINISHED = 4
+const SORT_ALPHABATICALLY = 1;
+const SORT_DEADLINE = 2;
+const SORT_NOTCOMPLETE = 3;
+const SORT_FINISHED = 4;
```

Comment:
great usage of constants.

consider using objects for those constants, for readability:
```suggestion
const SORT_TYPES = {
  ALPHABATICALLY: 1,
  DEADLINE: 2,
  ...
};
```